vnext/JSI/Shared/ChakraRuntimeArgs.h

```diff
@@ -2,49 +2,71 @@
 // Licensed under the MIT License.
 
 #pragma once
-#include <jsi/jsi.h>
-#include "ScriptStore.h"
+
+#include "PreparedJavaScriptStore.h"
+
+#include <cstddef>
+#include <memory>
+#include <optional>
+#include <utility>
 
 namespace facebook::react {
 class MemoryTracker;
-class MessageQueueThread;
-} // namespace facebook::react
+}
 
 namespace Microsoft::JSI {
 
-enum class LogLevel {
-  Trace = 0,
-  Info = 1,
-  Warning = 2,
-  Error = 3,
-  Fatal = 4,
-};
-
-using Logger = std::function<void(const char *message, LogLevel logLevel)>;
-
 struct ChakraRuntimeArgs {
-  bool enableJITCompilation{true};
+ public:
+  /**
+   * @enum ExecutablePageAllocationPolicy Indicates whether a ChakraRuntime
+   * is allowed to allocate executatble pages.
+   */
+  enum class ExecutablePageAllocationPolicy {
+    /**
+     * @var ChakraRuntime will not allocate any executable code pages. This
+     * also implies that Native Code generation will be turned off. Note that
+     * this will break JavaScript stack decoding in tools like WPA, since
+     * they rely on allocation of unique thunks to interpret each function and
+     * allocation of those thunks will be disabled as well
+     */
+    DisableAll,
+    /**
+     * @var ChakraRuntime will not generate native code.
+     */
+    DisableNativeCodeGeneration,
+    /**
+     * @var ChakraRuntime is free to generate executable pages.
+     */
+    EnableAll
+  };
 
-  Logger loggingCallback{};
+  /**
+   * @brief ChakraRuntime's policy for allocating executable pages.
+   */
+  ExecutablePageAllocationPolicy m_executablePageAllocationPolicy{ExecutablePageAllocationPolicy::EnableAll};
 
-  // Debugger configuration.
-  bool enableDebugging{false};
-  bool debuggerBreakOnNextLine{false};
-  bool enableNativePerformanceNow{true};
-  bool enableNativePromiseSupport{false};
-  uint16_t debuggerPort{9229};
-  std::string debuggerRuntimeName;
+  /**
+   * TODO (yicyao)
+   */
+  bool m_enableDebugging{false};
 
-  // Memory governance.
-  size_t runtimeMemoryLimit{0};
-  std::shared_ptr<facebook::react::MemoryTracker> memoryTracker;
+  /**
+   * @brief The maximum amount of memory the JavaScript engine may allocate,
+   * in bytes. Its value must be greater than zero.
+   */
+  std::optional<size_t> m_memoryLimit{};
 
-  std::shared_ptr<facebook::react::MessageQueueThread> jsQueue;
+  /**
+   * @brief The memory tracker that monitors ChakraRuntime's memory consumption.
+   * This argument is optional and can be nullptr.
+   */
+  std::shared_ptr<facebook::react::MemoryTracker> m_memoryTracker{nullptr};
 
-  // Script store which manages script and prepared script storage and
-  // versioning.
-  std::unique_ptr<facebook::jsi::ScriptStore> scriptStore;
-  std::unique_ptr<facebook::jsi::PreparedScriptStore> preparedScriptStore;
+  /**
+   * TODO (yicyao)
+   */
+  std::unique_ptr<PreparedJavaScriptStore> m_preparedJsStore{nullptr};
```

Comment:
>unique_ptr [](start = 7, length = 10)

Will we need to share the script store ever with something outside the runtime? Wondering if this should be ref counted.

Reply:
I am not sure whether we need ref counting here or not. Let's keep this a unique_ptr for now and change it to a shared_ptr if needed in the future.

---
In reply to: [345456324](https://github.com/microsoft/react-native-windows/pull/3650#discussion_r345456324) [](ancestors = 345456324)